### ISSUE TITLE
Input ordering is not explicitly stated

### ISSUE THREAD

**Issue by haakom** [lucidrains/performer-pytorch#35]
Since the pytorch implementation of a transformer expects input ordering to be (sequence_length, batch_size, feature_dimension) is this also the case for the performer implementation?

For instance, using the SelfAttention as such:
attn = SelfAttention(
    dim = 512,
    heads = 8,
    causal = False,
).cuda()

Should it be:
x = torch.randn(batch_size, sequence_length, feature_dimension).cuda()

or:
x = torch.randn(sequence_length, batch_size, feature_dimension).cuda()

attn(x)

?

**Comment by lucidrains** [comment]
@haakom ohh! sorry this wasn't clear

in all of my repositories, I assume `(batch x seq_len x feature_dim)`

**Comment by haakom** [comment]
@lucidrains That's great, thanks! 

That's what I assumed from reading the code, but it is nice to have it confirmed.

Thank you for the project! You've enabled me to train some models I've been wanting to try since the Transformers were first introduced.
